### PR TITLE
upgrade(package): Update drivelist 5.2.12 -> 6.0.0

### PR DESCRIPTION
**Pull request by jhermsmeier**
This updates `drivelist` to incorporate the new native
bindings rewrite on Windows, fixing scanning errors.

**TODO:**
- [x] Merge https://github.com/resin-io-modules/drivelist/pull/243
- [x] Update package.json & npm-shrinkwrap.json to the correct dependency

Change-Type: patch
Changelog Entry: Fix "Couldn't scan drives" error
Connects To: #1646, #1660, #1665, #1782, #1915, #1867, #1757

### CODE REVIEW

File: lib/shared/sdk/standard/index.js [hunk 1]
```diff
@@ -71,15 +71,21 @@ class StandardAdapter extends EventEmitter {
...
 
       return drive
     }).catch((error) => {
       this.emit('error', error)
       callback && callback(error)
     }).filter((drive) => {
-      return options.includeSystemDrives || !drive.system
+      return !drive.error && (options.includeSystemDrives || !drive.isSystem)
+    }).map((drive) => {
+      drive.displayName = drive.device
```
Comment (jviotti) [inline lib/shared/sdk/standard/index.js]: Can we move the displayName to `drivelist`, given we're likely going to use the same thing on the CLI? Of course doesn't need to be done right now

File: tests/shared/models/available-drives.spec.js [hunk 2]
```diff
@@ -354,9 +386,11 @@ describe('Model: availableDrives', function () {
               device: '/dev/sdc',
               name: 'USB Drive',
               size: 9999999,
-              mountpoint: '/mnt/bar',
-              system: false,
-              protected: false
+              mountpoints: [ {
```
Comment (jviotti) [inline tests/shared/models/available-drives.spec.js]: I have the impression the linter will complain about this.

Comment (jhermsmeier) [reply tests/shared/models/available-drives.spec.js]: Quite the opposite; it wants a space there... :/

File: lib/shared/sdk/standard/index.js (hunk 1 above)
Comment (jhermsmeier) [reply lib/shared/sdk/standard/index.js]: I was thinking that this logic is very application specific – in general the `description` is the best display name, imo, so the CLI might just want to switch to that?
I just recreated the displayName here, as we've got this fairly specific difference in handling them on Win in contrast to Linux / Mac in Etcher.
What do you think? Of course we can add it back to drivelist also, but I feel that it shouldn't be concerned with these UI specific things.

**Comment by jviotti** [comment]
We should unify how drives are displayed in both the GUI and the CLI
version. If the description is the way to go (and we determine so in ie
a product call), then we should use the description everywhere.
Otherwise, we rather put the logic in one place and re-use a new propery
in both UIs.

I think providing a "display name" property in drivelist is fine, and
can be ignored if people want to use something else.

On Fri, Jan 05, 2018 at 03:56:50PM +0000, Jonas Hermsmeier wrote:
> jhermsmeier commented on this pull request.
> 
> 
> 
> >        }
>  
>        return drive
>      }).catch((error) => {
>        this.emit('error', error)
>        callback && callback(error)
>      }).filter((drive) => {
> -      return options.includeSystemDrives || !drive.system
> +      return !drive.error && (options.includeSystemDrives || !drive.isSystem)
> +    }).map((drive) => {
> +      drive.displayName = drive.device
> 
> I was thinking that this logic is very application specific – in general the `description` is the best display name, imo, so the CLI might just want to switch to that?
> I just recreated the displayName here, as we've got this fairly specific difference in handling them on Win in contrast to Linux / Mac in Etcher.
> What do you think? Of course we can add it back to drivelist also, but I feel that it shouldn't be concerned with these UI specific things.
> 
> -- 
> You are receiving this because your review was requested.
> Reply to this email directly or view it on GitHub:
> https://github.com/resin-io/etcher/pull/1953#discussion_r159909635

-- 
Juan Cruz Viotti
Software Engineer

File: tests/gui/modules/drive-scanner.spec.js [hunk 3]
```diff
@@ -332,7 +367,8 @@ describe('Browser: driveScanner', function () {
                 path: 'H:'
               }
             ],
-            system: false
+            isSystem: false,
+            isRemovable: true
```
Comment (lurch) [inline tests/gui/modules/drive-scanner.spec.js]: I'm wondering why you've added `isRemovable` to all these test-cases, given that nothing else even looks at this property?

**Comment by lurch** [comment]
I can see both points of view, and from a "purist" PoV `displayName` doesn't belong in `drivelist`; but I think I tend to agree with @jviotti 's pragmatist PoV that it's such a generally-useful property that it makes sense to add it back to `drivelist`.

**Pull request merged**